tools/build-tools/src/common/fluidRepoBase.ts

```diff
@@ -6,25 +6,58 @@
 import * as path from "path";
 import { Package, Packages } from "./npmPackage";
 import { MonoRepo, MonoRepoKind } from "./monoRepo";
+import { FluidPackageCheck } from "./build/fluidPackageCheck";
+import { globFn } from "./utils";
+import { NpmDepChecker } from "./build/npmDepChecker";
+import { ISymlinkOptions, symlinkPackage } from "./build/symlinkUtils";
+import { BuildGraph } from "./build/buildGraph";
+import { logVerbose } from "./logging";
+
+export interface IPackageMatchedOptions {
```

Comment:
My thought process here was that since they were both common necessities of a repo, it might make sense to merge it into one and instead let the calling script decide how to use the class?
i.e. I feel like having the build capabilities in the base class is helpful to anybody who will be calling the class APIs. Then the FluidBuild script can decide, based on the parameters in package.json, how to go about navigating the repo and using the API functions accordingly. Since we get the specification granularity through json, I was thinking that we might not need a separate class definition now to store that, along with some more build functions separately.
I suppose it relies on how "much" we expect "building" to be a core tenet of being a "repo". I'm totally open to keeping the APIs separated though, let me know!